### PR TITLE
new handler to catch and report panics

### PR DESCRIPTION
**Pull request by csquared**
add a main-package level `recoverWith(f func(err error))` function you can use with defer to recover from panics.

installs the handler in the web, cluster monitor, and heartbeat processes. each one of these is performing a stateless task in a loop, so it makes sense to just report the error and keep processing while we can decide how to intervene.

i did not install this in the pull app images goroutine b.c that seemed to only run on boot and has its own retry logic.

### CODE REVIEW

**Comment by csquared** [comment]
I left the `/panic` endpoint in place so we dont’ have to figure out how to cause a panic

I’ve called those /boom in the past.  They're also good to make sure stuff like alerting is hooked up. Thoughts?

File: helpers/helpers.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ func init() {
 }
 
 func Error(log *logger.Logger, err error) {
-	if log == nil {
+	if log != nil {
```
Comment (csquared) [inline helpers/helpers.go]: this is a bugfix

**Comment by csquared** [comment]
I went with "boom"

**Comment by ddollar** [comment]
:+1:

**Pull request merged**